### PR TITLE
Constants for exit codes

### PR DESCRIPTION
**Pull request by zilm13**
## Fixed Issue(s)

I want to organize exit codes. From my understanding current agreement is to use code 2 when restart will not help, 1 when it could help.

- `infrastructure:exceptions` looks like a good place for it, we already have FatalServiceException there and it will not toggle module import hell
- I've changed some current values to follow this agreement, not sure it was a right decision in all cases
- https://github.com/ConsenSys/teku/blob/f31b239f7788f07d8e9c2b6e7544c5e16b01a259/teku/src/main/java/tech/pegasys/teku/Teku.java#L57 return codes which could fire an exit is not covered, though it returns codes in line with our agreement. I could make separate PR to override them explicitly if needed

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/ExportCommand.java [hunk 1]
```diff
@@ -69,13 +71,14 @@ public void run() {
...
-          1, "There were errors reading from slashing protection files, cannot complete.");
+          FATAL_EXIT_CODE,
+          "There were errors reading from slashing protection files, cannot complete.");
     }
     try {
       SUB_COMMAND_LOG.display("Writing slashing protection data to: " + toFileName);
       slashingProtectionExporter.saveToFile(toFileName, SUB_COMMAND_LOG::display);
     } catch (IOException e) {
-      SUB_COMMAND_LOG.exit(1, "Failed to export slashing protection data.", e);
+      SUB_COMMAND_LOG.exit(FATAL_EXIT_CODE, "Failed to export slashing protection data.", e);
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/ExportCommand.java]: in export command it's not a service, these are probably less important... i'd probably leave them at code 1, but I don't feel super strongly
My logic here is with a CLI tool a 2 literally indicates you've used it wrongly... so like you've given me an option that isn't compatible...

File: teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/ImportCommand.java [hunk 2]
```diff
@@ -65,11 +67,12 @@ public void run() {
...
       SUB_COMMAND_LOG.display("Reading slashing protection data from: " + importFile);
       final Optional<String> errorCondition = importer.initialise(importFile);
-      errorCondition.ifPresent(s -> SUB_COMMAND_LOG.exit(1, s));
+      errorCondition.ifPresent(s -> SUB_COMMAND_LOG.exit(FATAL_EXIT_CODE, s));
     } catch (IOException e) {
       String cause = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
       SUB_COMMAND_LOG.exit(
-          1, String.format("Failed to read from import file: %s. %s", importFile, cause));
+          FATAL_EXIT_CODE,
+          String.format("Failed to read from import file: %s. %s", importFile, cause));
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/ImportCommand.java]: i'd probalby argue the same for all the tools, i'd leave them at a 1 unless it truly is a usage issue. maybe an exit code for usage if we ever actually do have that scenario...

File: teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/RepairCommand.java [hunk 3]
```diff
@@ -133,7 +136,7 @@ private UInt64 getComputedSlot(final Optional<AnchorPoint> initialAnchor, final
       return computedSlot;
     }
     SUB_COMMAND_LOG.exit(
-        1,
+        FATAL_EXIT_CODE,
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/slashingprotection/RepairCommand.java]: same here.

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/MigrateDatabaseCommand.java [hunk 4]
```diff
@@ -122,14 +125,14 @@ public void run() {
     // validate output format
     final Optional<DatabaseVersion> maybeOutputVersion = DatabaseVersion.fromString(toDbVersion);
     if (maybeOutputVersion.isEmpty()) {
-      SUB_COMMAND_LOG.exit(2, "Invalid database version specified: " + toDbVersion);
+      SUB_COMMAND_LOG.exit(FATAL_EXIT_CODE, "Invalid database version specified: " + toDbVersion);
     }
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/MigrateDatabaseCommand.java]: all commands really i'll stop commenting maybe...

File: teku/src/main/java/tech/pegasys/teku/cli/util/SlashingProtectionCommandUtils.java [hunk 5]
```diff
@@ -24,7 +26,7 @@ public static void verifySlashingProtectionPathExists(
       final SubCommandLogger subCommandLogger, final Path slashProtectionPath) {
     if (!slashProtectionPath.toFile().exists() || !slashProtectionPath.toFile().isDirectory()) {
       subCommandLogger.exit(
-          1,
+          FATAL_EXIT_CODE,
```
Comment (rolfyone) [inline teku/src/main/java/tech/pegasys/teku/cli/util/SlashingProtectionCommandUtils.java]: actually i think this is the last command tool, so may as well say this one too...

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java [hunk 6]
```diff
@@ -284,7 +285,7 @@ private static void checkNoKeysLoaded(
     if (validatorConfig.isExitWhenNoValidatorKeysEnabled()
         && validatorLoader.getOwnedValidators().hasNoValidators()) {
       STATUS_LOG.exitOnNoValidatorKeys();
-      System.exit(2);
+      System.exit(FATAL_EXIT_CODE);
```
Comment (rolfyone) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java:294]: i do like the look of this more, i'll admit.

**Comment by rolfyone** [comment]
lots of comments but TL/DR, i think this is the right direction to take, just that command line tools i'd leave out of the change, or leave them at code 1 equiv unless it's truly a usage issue (generally should be picocli returning 2)

**Comment by zilm13** [comment]
@rolfyone I ended up removing it from the commands usage and adding a comment that it's not for commands

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**